### PR TITLE
TTL and client beat interval should be customized at instance level.

### PR DESCRIPTION
**Pull request by IanCao**
Please do not create a Pull Request without creating an issue first.

## What is the purpose of the change

XXXXX

## Brief changelog

XX

## Verifying this change

XXXX

Follow this checklist to help us incorporate your contribution quickly and easily:

* [ ] Make sure there is a Github issue filed for the change (usually before you start working on it). Trivial changes like typos do not require a Github issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.
* [ ] Format the pull request title like `[ISSUE #123] Fix UnknownException when host config not exist`. Each commit in the pull request should have a meaningful subject line and body.
* [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
* [ ] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add integration-test in [test module](https://github.com/alibaba/nacos/tree/master/test).
* [ ] Run `mvn -B clean package apache-rat:check findbugs:findbugs -Dmaven.test.skip=true` to make sure basic checks pass. Run `mvn clean install -DskipITs` to make sure unit-test pass. Run `mvn clean test-compile failsafe:integration-test`  to make sure integration-test pass.

### CODE REVIEW

**Comment by IanCao** [comment]
#1353

**Comment by nkorange** [comment]
@IanCao Please use github account to push your commits

**Comment by IanCao** [comment]
> @IanCao Please use github account to push your commits

I have changed my github account

File: api/src/main/java/com/alibaba/nacos/api/common/Constants.java [hunk 1]
```diff
@@ -42,6 +44,12 @@
 
     public static final String GROUP = "group";
 
+    public static final String HEART_BEAT_TIMEOUT = "heartBeatTimeout";
```
Comment (nkorange) [inline api/src/main/java/com/alibaba/nacos/api/common/Constants.java]: Please define these keys in class PreservedMetadataKeys and the keys should start with 'preserved.'

Comment (nkorange) [inline api/src/main/java/com/alibaba/nacos/api/common/Constants.java]: Please define these keys in class PreservedMetadataKeys and the keys should start with 'preserved.'

Comment (IanCao) [reply api/src/main/java/com/alibaba/nacos/api/common/Constants.java]: ok

Comment (IanCao) [reply api/src/main/java/com/alibaba/nacos/api/common/Constants.java]: ok

File: client/src/main/java/com/alibaba/nacos/client/naming/beat/BeatReactor.java [hunk 2]
```diff
@@ -76,22 +76,24 @@ public String buildKey(String serviceName, String ip, int port) {
...
     class BeatProcessor implements Runnable {
-
         @Override
         public void run() {
-            try {
-                for (Map.Entry<String, BeatInfo> entry : dom2Beat.entrySet()) {
-                    BeatInfo beatInfo = entry.getValue();
-                    if (beatInfo.isScheduled()) {
-                        continue;
+            while (true) {
```
Comment (nkorange) [inline client/src/main/java/com/alibaba/nacos/client/naming/beat/BeatReactor.java]: Should not use 'while(true)', consider when executorService's pool has only one thread.

File: client/src/main/java/com/alibaba/nacos/client/naming/NacosNamingService.java [hunk 3]
```diff
@@ -193,13 +196,28 @@ public void registerInstance(String serviceName, String groupName, Instance inst
...
+            long instanceInterval = getInstanceHeartBeatInterval(instance);
+            beatInfo.setPeriod(instanceInterval == 0 ? DEFAULT_HEART_BEAT_INTERVAL : instanceInterval);
 
             beatReactor.addBeatInfo(NamingUtils.getGroupedName(serviceName, groupName), beatInfo);
         }
 
         serverProxy.registerService(NamingUtils.getGroupedName(serviceName, groupName), groupName, instance);
     }
 
+    private long getInstanceHeartBeatInterval(Instance instance) {
```
Comment (nkorange) [inline client/src/main/java/com/alibaba/nacos/client/naming/NacosNamingService.java]: This method should be placed in class 'Instance'. So are the methods in 'ClientBeatCheckTask'.

**Comment by nkorange** [comment]
Add necessary test cases please.

File: client/src/main/java/com/alibaba/nacos/client/naming/NacosNamingService.java (hunk 3 above)
Comment (IanCao) [reply client/src/main/java/com/alibaba/nacos/client/naming/NacosNamingService.java]: ok

File: client/src/main/java/com/alibaba/nacos/client/naming/beat/BeatReactor.java (hunk 2 above)
Comment (IanCao) [reply client/src/main/java/com/alibaba/nacos/client/naming/beat/BeatReactor.java]: i cant get ur point  about only has one thread.

File: client/src/main/java/com/alibaba/nacos/client/naming/beat/BeatInfo.java [hunk 4]
```diff
@@ -31,6 +31,8 @@
     private String cluster;
     private Map<String, String> metadata;
     private volatile boolean scheduled;
+    private volatile long period;
+    private volatile boolean stop;
```
Comment (nkorange) [inline client/src/main/java/com/alibaba/nacos/client/naming/beat/BeatInfo.java]: Use 'stopped'

Comment (IanCao) [reply client/src/main/java/com/alibaba/nacos/client/naming/beat/BeatInfo.java]: i have updated

**Pull request merged**